text/0000-thread-safe-handles.md

```diff
@@ -0,0 +1,362 @@
+- Feature Name: Threadsafe Handles
+- Start Date: 2020-07-21
+- RFC PR: (leave this empty)
+- Neon Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Two new `Send` and `Sync` features are introduced: `EventQueue: Send + Sync` and `Persistent: Send`. These features work together to allow multi-threaded modules to interact with the JavaScript main thread.
+
+* `EventQueue`: Threadsafe handle that allows sending closures to be executed on the main thread.
+* `Persistent<T>`: Opaque handle that can be sent across threads, with inner contents only accessible on the main thread.
```

Comment:
Another word for this might be `Shared<T>`. It's a little less cumbersome and daunting, and carries the intuition of shared state.

Reply:
I like `Shared`. Do you happen to know if Rust has an analogous type? I.e., a type that can be sent across threads, but only dereferenced on the original thread? It probably doesn't; it doesn't feel very rust-y.